java/source/flight.rst

```diff
@@ -0,0 +1,485 @@
+                            listArrowRecordBatch.add(arb);
+                            rows += flightStream.getRoot().getRowCount();
+                        }
+                    }
+                    DataInMemory pojoFlightDataInMemory = new DataInMemory(listArrowRecordBatch, flightStream.getSchema(), rows);
+                    dataInMemory.put(flightStream.getDescriptor(), pojoFlightDataInMemory);
+                    ackStream.onCompleted();
+                };
+            }
+
+            @Override
+            public void getStream(CallContext context, Ticket ticket, ServerStreamListener listener) {
+                FlightDescriptor flightDescriptor = FlightDescriptor.path(new String(ticket.getBytes(), StandardCharsets.UTF_8)); // Recover data for key configured
+                if(dataInMemory.containsKey(flightDescriptor)){
+                    VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(dataInMemory.get(flightDescriptor).getSchema(), allocator);
+                    listener.start(vectorSchemaRoot);
+                    for(ArrowRecordBatch arrowRecordBatch : dataInMemory.get(flightDescriptor).getListArrowRecordBatch()){
+                        VectorLoader loader = new VectorLoader(vectorSchemaRoot);
+                        loader.load(arrowRecordBatch.cloneWithTransfer(allocator));
+                        listener.putNext();
+                    }
+                    listener.completed();
+                }
+            }
+
+            @Override
+            public void doAction(CallContext context, Action action, StreamListener<Result> listener) {
+                FlightDescriptor flightDescriptor = FlightDescriptor.path(new String(action.getBody(), StandardCharsets.UTF_8)); // For recover data for key configured
+                if(dataInMemory.containsKey(flightDescriptor)) {
+                    switch (action.getType()) {
+                        case "DELETE":
+                            dataInMemory.remove(flightDescriptor);
+                            Result result = new Result("Delete completed".getBytes(StandardCharsets.UTF_8));
+                            listener.onNext(result);
+                    }
+                    listener.onCompleted();
+                }
+            }
+
+            @Override
+            public FlightInfo getFlightInfo(CallContext context, FlightDescriptor descriptor) {
+                if(!dataInMemory.containsKey(descriptor)){
+                    throw CallStatus.NOT_FOUND.withDescription("Unknown descriptor").toRuntimeException();
+                }
+                return new FlightInfo(
+                        dataInMemory.get(descriptor).getSchema(),
+                        descriptor,
+                        Collections.singletonList(new FlightEndpoint(new Ticket(descriptor.getPath().get(0).getBytes(StandardCharsets.UTF_8)), location)), // Configure a key to map back and forward your data using Ticket argument
```

Comment:
Can we move the FlightEndpoint onto its own line?

Reply:
Moved